m4/ax_check_botan.m4

```diff
@@ -130,4 +149,5 @@ AC_DEFUN([AX_CHECK_BOTAN_DEFINES], [
             $2
         ],
         [#include <botan/build.h>])
+    CPPFLAGS="$save_CPPFLAGS"
```

Comment:
1. I would say it's bad behavior for an AC function to modify CPPFLAGS/etc.
2. I much prefer to keep things fine-grained and not set things like that globally. Not all of our makefile targets require botan headers.